pkg/util/podstore/podstore.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podstore
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/clock"
+)
+
+const (
+	deleteQueueName = "pod_delete"
+	podIPIndex      = "PodIP"
+	delayTime       = time.Minute * 5
+)
+
+type PodStore struct {
+	pods         cache.Indexer
+	podsToDelete workqueue.DelayingInterface
+	// Mapping pod.uuid to PodLocalTimestamps
+	timestampMap map[types.UID]*podLocalTimestamps
+	clock        clock.Clock
+	mutex        sync.Mutex
+}
+
+type podLocalTimestamps struct {
```

Comment:
let's rename this to `podTimestamps` now that we also rely on `pod.CreationTimestamp`

Reply:
Updated.